packages/ui-kit-components/src/components/TextSearch/TextSearch.tsx

```diff
@@ -0,0 +1,91 @@
+import { Autocomplete, TextField } from '@mui/material';
+
+import * as Icons from '../../icons';
+
+export interface TextSearchProps<T> {
```

Comment:
I'll see if it can be done. The AutocompeteProps type is complex, having 4 additional generics. I think I'll just need to figure out what the right types are for them and Omit any properties that shouldn't be used.